docs: Align 'Configuring Runtime Watcher' setup guide with implementation (#712)
kyma-project/runtime-watcher: docs/watcher-setup-guide.md

To set up a watch on a resource, you must define and apply a Watcher CR for it. The Watcher CR defines which resources Runtime Watcher notifies changes for and where to forward the events in KCP.

> Here is the definition of the Watcher CR. The detailed field descriptions are provided below in the [Resources to Watch](#resources-to-watch) and [Events to Consume](#events-to-consume) sections.

Reviewer: Thanks for the cleanup, Two things worth aligning before merge:

- The `ReceivedEvents()` doc comment lists `watched-gvk` (hyphenated) but the guide uses `watchedGvk` (camelCase). Could trip up someone using the listener package directly.
- `api.md` describes the path as `/validate/<manager-value>` while the guide says `/v2/<manager>/event`. These seem to describe the SKR-side webhook path vs the KCP-side route respectively — worth a clarifying note so readers aren't confused by the two different paths.

Author: > The ReceivedEvents() doc comment lists watched-gvk (hyphenated) but the guide uses watchedGvk (camelCase). Could trip up someone using the listener package directly.

The latter is describing the JSON payload where we use `watchedGvk`: https://github.com/kyma-project/runtime-watcher/blob/de2f534ce7c0c73da817505c9aad0db12f966b27/listener/pkg/v2/types/event.go#L13. The former describes the listener where we add `watched-gvk` to the unstructured object: https://github.com/kyma-project/runtime-watcher/blob/de2f534ce7c0c73da817505c9aad0db12f966b27/listener/pkg/v2/event/watcher_event.go#L95

> api.md describes the path as /validate/<manager-value> while the guide says /v2/<manager>/event. These seem to describe the SKR-side webhook path vs the KCP-side route respectively — worth a clarifying note so readers aren't confused by the two different paths.

Valid. We also had info duplicated between the setup and api docs. I moved everything to the api doc now and only link to the explanations there: https://github.com/kyma-project/runtime-watcher/pull/712/commits/1a34f58adf1f10586c6c8390b27e1f6d127023a2.

To set up a watch on a resource, you must define and apply a Watcher CR for it. The Watcher CR defines which resources Runtime Watcher notifies changes for and where to forward the events in KCP.

Here is an example of the Watcher CR. The detailed field descriptions are provided in the [Watcher API definition](./api.md).

```yaml
apiVersion: operator.kyma-project.io/v1beta2
kind: Watcher
metadata:
  name: <name>
  namespace: kcp-system
spec:
  resourceToWatch:
    group: <api-group>
    version: <version>
    resource: <kind>
  labelsToWatch:
    "<some>": "<label>"
  field: <"spec" or "status">
  manager: <manager-name>
  serviceInfo:
    name: <service-name>
    port: <port>
    namespace: <namespace>
  gateway: # don't change
    selector:
      matchLabels:
        "operator.kyma-project.io/watcher-gateway": "default"
```

## Consuming Events

> The service receiving the events can be any arbitrary service that is listening on the specified port. Behind the service there needs to be consumer expecting POST requests on `/v2/<spec.manager>/event` with the following content:

Reviewer 2: ```suggestion
The service receiving the events can be any arbitrary service that is listening on the specified port. Behind the service, there must be a consumer expecting POST requests on `/v2/<spec.manager>/event` with the following content:
```

```json
{
  "watched": { "Namespace": "<watched object's namespace>", "Name": "<watched object's name>" },
  "watchedGvk": { "group": "<watched object's group>", "version": "<watched object's version>", "kind": "<watched object's kind>" }
}
```

> To determine what Kyma Runtime the received event is from, the Runtime Id can be extracted from the Common Name of the certificate attached to the request. The certificate attached to the request is available as HTTP header and the `listener` package provides the [`GetCertificateFromHeader()`](https://github.com/kyma-project/runtime-watcher/blob/de2f534ce7c0c73da817505c9aad0db12f966b27/listener/pkg/v2/certificate/parse_certificate.go#L26-L65) helper function to extract it. It can be used as follows:

Reviewer 2: ```suggestion
To identify the Kyma runtime from which the received event originates, the Runtime Id can be extracted from the Common Name of the certificate attached to the request. The certificate attached to the request is available as an HTTP header, and the `listener` package provides the [`GetCertificateFromHeader()`](https://github.com/kyma-project/runtime-watcher/blob/de2f534ce7c0c73da817505c9aad0db12f966b27/listener/pkg/v2/certificate/parse_certificate.go#L26-L65) helper function to extract it. It can be used as follows:
```

```Go
func getRuntimeIdFromRequest(req *http.Request) (string, *UnmarshalError) {
	clientCertificate, err := certificate.GetCertificateFromHeader(req)
	if err != nil {
		return "", &UnmarshalError{
			fmt.Sprintf("could not get client certificate from request: %v", err),
			http.StatusUnauthorized,
		}
	}

	if clientCertificate.Subject.CommonName == "" {
		return "", &UnmarshalError{
			"client certificate common name is empty",
			http.StatusBadRequest,
		}
	}

	return clientCertificate.Subject.CommonName, nil
}
```

> For further convenience, the `listener` package also provides a [`SKREventListener`](../listener/pkg/v2/event/skr_events_listener.go#L32-L43) that handles the requests and exposes a channel via [`ReceivedEvents()`](../listener/pkg/v2/event/skr_events_listener.go#L46-L54) providing an unstructured object for every received event. For a usage example, refer to Lifecycle Manager:

Reviewer 2: ```suggestion
For further convenience, the `listener` package also provides a [`SKREventListener`](../listener/pkg/v2/event/skr_events_listener.go#L32-L43) that handles the requests and exposes a channel via [`ReceivedEvents()`](../listener/pkg/v2/event/skr_events_listener.go#L46-L54) providing an unstructured object for every received event. For an example in Lifecycle Manager, see:
```
